Content.Server/_DV/Abilities/Kitsune/KitsuneSystem.cs

```diff
@@ -0,0 +1,138 @@
+        {
+            _actionsSystem.AddAction(uid, ref component.KitsuneActionEntity, component.KitsuneAction);
+        }
+    }
+
+    private void OnMorphIntoKitsune(EntityUid humanoidUid, KitsuneComponent component, MorphIntoKitsune args)
+    {
+
+        var foxUid = _polymorphSystem.PolymorphEntity(humanoidUid, component.KitsunePolymorphId);
+
+        if (!foxUid.HasValue)
+            return;
+
+        if (TryComp<AppearanceComponent>(foxUid, out var appearanceComp))
+        {
+            _appearance.SetData(foxUid.Value, KitsuneColor.Color, _eyeColor ?? Color.Orange, appearanceComp);
+        }
+
+        //Transfer Accesses
+        var accessItems = _reader.FindPotentialAccessItems(humanoidUid);
+        var accesses = _reader.FindAccessTags(humanoidUid, accessItems);
+        EnsureComp<AccessComponent>((EntityUid)foxUid);
+        _access.TrySetTags((EntityUid)foxUid, accesses);
+
+        //Transfer factions
+        if (TryComp<NpcFactionMemberComponent>(humanoidUid, out var factions))
+        {
+            EnsureComp<NpcFactionMemberComponent>((EntityUid)foxUid);
+            _faction.AddFactions((EntityUid)foxUid, factions.Factions);
+        }
+
+        _popupSystem.PopupEntity(Loc.GetString("kitsune-popup-morph-message-others", ("entity", foxUid.Value)), foxUid.Value, Filter.PvsExcept(foxUid.Value), true);
+        _popupSystem.PopupEntity(Loc.GetString("kitsune-popup-morph-message-user"), foxUid.Value, foxUid.Value);
+
+        args.Handled = true;
+    }
+    private void OnCreateFoxfire(EntityUid uid, KitsuneComponent component, CreateFoxfireActionEvent args)
+    {
+        if (!TryComp<HandsComponent>(uid, out var hands) || hands.Count < 1)
+        {
+            _popupSystem.PopupEntity(Loc.GetString("fox-no-hands"), uid, uid);
+            return;
+        }
+
+        if (_actions.GetCharges(component.FoxfireAction) is { } and < 1)
+        {
+            _entities.DeleteEntity(component.ActiveFoxFires[0]);
+            component.ActiveFoxFires.RemoveAt(0);
+        }
+
+        if (_actions.GetCharges(component.FoxfireAction) is { } and <= -1)
+        {
+            _actions.SetCharges(component.FoxfireAction, 1);
+        }
+        var fireEnt = Spawn(component.FoxfirePrototype, Transform(uid).Coordinates);
+        var fireComp = EnsureComp<FoxFireComponent>(fireEnt);
+        fireComp.Owner = uid;
+        component.ActiveFoxFires.Add(fireEnt);
+
+        if (_eyeColor is not null)
+            _light.SetColor(fireEnt, (Color)_eyeColor);
+
+        args.Handled = true;
+    }
+    private void OnFoxFireShutdown(EntityUid uid, FoxFireComponent component, ComponentShutdown args)
+    {
+        if (component.Owner is null)
+            return;
+        RaiseLocalEvent<FoxfireDestroyedEvent>(component.Owner.Value, new());
+    }
+    private void OnFoxFireDestroyed(EntityUid uid, KitsuneComponent component, FoxfireDestroyedEvent args)
+    {
+        component.ActiveFoxFires.Remove(uid);
```

Comment:
`Dirty(ent);` afterwards, this code also didnt work as it used the kitsune entity instead of the foxfire